File: 3567-convert-date-to-binary/convert-date-to-binary.py
class Solution:
    def convertDateToBinary(self, date: str) -> str:
        # date = list(date.split('-'))
        # for i in range(len(date)):
        #     date[i] = bin(int(date[i]))[2:]
        # return '-'.join(date)

        parts = date.split('-')
        binary_parts = [bin(int(part))[2:] for part in parts]
        return '-'.join(binary_parts)

File: 3567-convert-date-to-binary/convert-date-to-binary.js
/**
 * @param {string} date
 * @return {string}
 */
var convertDateToBinary = function(date) {
    const date_parts = date.split('-');
    for (let i = 0; i < date_parts.length; i++){
        let part_num = parseInt(date_parts[i]);
        date_parts[i] = part_num.toString(2)
    }
    return date_parts.join('-')
};